include/envoy/network/filter.h

```diff
@@ -81,8 +81,7 @@ class ReadFilter {
   /**
    * Called when data is read on the connection.
    * @param data supplies the read data which may be modified.
-   * @param end_stream supplies whether this is the last byte on the connection. This will only
-   *        be set if the connection has half-close semantics enabled.
+   * @param end_stream supplies whether this is the last byte on the connection.
```

Comment:
Assuming we move forward with this change, can we clarify the docs here? I think this needs a pretty full treatment in terms of the difference between half-closed and not half-closed mode, and the fact that the "last byte" might be an empty data frame depending on the interleaving of data and FIN.